### PR TITLE
Hackstack page

### PR DESCRIPTION
**Pull request by gabriellui1**
fixes ENG-1248
Waiting for the image to be added in the "Create" section and the completion of the FAQ.
hackstack-page

## Release Notes

- **New Features**
  - Introduced a new landing page for the HackStack initiative, featuring components for FAQs, safety information, professional development, and interactive learning sections.
  - Added dark mode support for additional paths, enhancing user experience based on context.
  - Implemented a module for AI HackStack, including learning, creating, and remixing projects with AI tools.

- **Enhancements**
  - Improved styling and layout for several components to support a cohesive visual theme.
  - Expanded routing capabilities to include a new path for HackStack, ensuring seamless navigation.

- **Documentation**
  - Updated localized content to include new features related to AI integration and user safety.

### CODE REVIEW

**Review by adamkecskes — COMMENTED** [review]
Looks great! Left a few comments + liked some coderabbit ones.

File: app/views/landing-pages/hackstack/SafetyComponent.vue [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+                >
+                <div class="div">
+                  <p class="header">
+                    {{ $t('hackstack_page.moderation_title') }}
+                  </p>
+                  <p class="text">
+                    {{ $t('hackstack_page.moderation_description') }}
+                  </p>
+                  <learn-more-button
+                    link="https://codecombat.com/safety"
```
[the remark this answers is not in the document]
Comment (mrfinch) [reply app/views/landing-pages/hackstack/SafetyComponent.vue:57]: @gabriellui1 u can use this link for now: https://docs.google.com/document/d/1OfQw0B841BUu7iABRlS-UXMbHDb9npYLvOPgcsdIiH0/edit?usp=sharing

File: app/views/landing-pages/hackstack/PDComponent.vue [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  </page-section>
+</template>
+
+<script>
+import PageSection from 'app/components/common/elements/PageSection'
+import ImageAndText from 'app/components/common/elements/ImageAndText.vue'
+import CTAButton from 'app/components/common/buttons/CTAButton.vue'
+
+export default {
+  name: 'SafeAIUse',
```
[the remark this answers is not in the document]
Comment (mrfinch) [reply app/views/landing-pages/hackstack/PDComponent.vue:31]: @gabriellui1 this needs to be fixed - name of file or component

**Review by mrfinch — APPROVED** [review]
Created a linear for minor comments mentioned here, great work on this one @gabriellui1

**Pull request merged**